pkg/sql/schemachanger/scbuild/build.go

```diff
@@ -62,12 +63,18 @@ func Build(
 		SchemaFeatureChecker: dependencies.FeatureChecker(),
 	}
 	defer func() {
-		if recErr := recover(); recErr != nil {
-			if errObj, ok := recErr.(error); ok {
-				err = errObj
-			} else {
-				err = errors.Errorf("unexpected error encountered while building schema change plan %s", recErr)
-			}
+		switch recErr := recover().(type) {
+		case nil:
+			// No error.
+		case runtime.Error:
+			err = errors.WithAssertionFailure(recErr)
```

Comment:
The nice thing about this is that it gives us a stack trace in the case of a nil pointer panic.